Allow optionals in builders.

ldstreet opened the pull request.

kyleve commented:
Mind updating the changelog as well?

kyleve commented:
Also seeing some test failures – should make sure we can make this change source compatible

ldstreet commented:
> Also seeing some test failures – should make sure we can make this change source compatible

Fixed this. I had some extra code that I removed locally but failed to push up. I think it should pass now.
